src/PositionContext.ts

```diff
@@ -439,32 +439,30 @@ export class PositionContext {
         return this._variableDefinition;
     }
 
-    private static getFunctionCompletions(prefix: string, replaceSpan: language.Span): Promise<Completion.Item[]> {
-        let functionMetadataPromise: Promise<FunctionMetadata[]>;
+    private static async getFunctionCompletions(prefix: string, replaceSpan: language.Span): Promise<Completion.Item[]> {
+        let functionMetadataMatches: FunctionMetadata[];
         if (prefix === "") {
-            functionMetadataPromise = AzureRMAssets.getFunctionMetadata();
+            functionMetadataMatches = (await AzureRMAssets.getFunctionMetadata()).functionMetadata;
         }
         else {
-            functionMetadataPromise = AzureRMAssets.getFunctionMetadataWithPrefix(prefix);
+            functionMetadataMatches = (await AzureRMAssets.getFunctionMetadataFromPrefix(prefix));
         }
 
-        return functionMetadataPromise.then((functionMetadataMatches: FunctionMetadata[]) => {
-            const completionItems: Completion.Item[] = [];
-            for (const functionMetadata of functionMetadataMatches) {
-                const name: string = functionMetadata.name;
+        const completionItems: Completion.Item[] = [];
+        for (const functionMetadata of functionMetadataMatches) {
+            const name: string = functionMetadata.name;
 
-                let insertText: string = name;
-                if (functionMetadata.maximumArguments === 0) {
-                    insertText += "()$0";
-                }
-                else {
-                    insertText += "($0)";
-                }
-
-                completionItems.push(new Completion.Item(name, insertText, replaceSpan, `(function) ${functionMetadata.usage}`, functionMetadata.description, Completion.Type.Function));
+            let insertText: string = name;
+            if (functionMetadata.maximumArguments === 0) {
+                insertText += "()$0";
+            }
+            else {
+                insertText += "($0)";
```

Comment:
Magic constant? :P

Reply:
No, it has to do with VS Code's completion contract, although I don't actually know what it means.

Reply:
You'll see I only changed indentation in these lines.